contracts/cw-proposal-single/src/contract.rs

```diff
@@ -726,9 +728,71 @@ pub fn query_info(deps: Deps) -> StdResult<Binary> {
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Don't do any state migrations.
-    Ok(Response::default())
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    /// This is proposal version is from commit e531c760a5d057329afd98d62567aaa4dca2c96f (v1.0.0) and code ID 427.
+    #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+    struct BetaProposal {
+        pub title: String,
+        pub description: String,
+        pub proposer: Addr,
+        pub start_height: u64,
+        pub min_voting_period: Option<Expiration>,
+        pub expiration: Expiration,
+        pub threshold: Threshold,
+        pub total_power: Uint128,
+        pub msgs: Vec<CosmosMsg<Empty>>,
+        pub status: Status,
+        pub votes: Votes,
+        pub allow_revoting: bool,
+        pub deposit_info: Option<CheckedDepositInfo>,
+    }
+
+    match msg {
+        MigrateMsg::FromBeta {} => {
+            // Retrieve current map from storage
+            let current_map: Map<u64, BetaProposal> = Map::new("proposals");
+            let current = current_map
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<(u64, BetaProposal)>>>()?;
+
+            // Add migrated entries to new map.
+            // Based on gas usage testing, we estimate that we will be able to migrate ~4200
+            // proposals at a time before reaching the block max_gas limit.
+            current
+                .into_iter()
+                .try_for_each::<_, StdResult<()>>(|(id, prop)| {
+                    let migrated_proposal = SingleChoiceProposal {
+                        title: prop.title,
+                        description: prop.description,
+                        proposer: prop.proposer,
+                        start_height: prop.start_height,
+                        min_voting_period: prop.min_voting_period,
+                        expiration: prop.expiration,
+                        threshold: prop.threshold,
+                        total_power: prop.total_power,
+                        msgs: prop.msgs,
+                        status: prop.status,
+                        votes: prop.votes,
+                        allow_revoting: prop.allow_revoting,
+                        deposit_info: prop.deposit_info,
+                        // AFAIK, Cosmwasm does not expose a way to query the timestamp
+                        // of a block given block height.
+                        // We assign migrated proposals a created timestamp of 0 so that the frontend may
+                        // query for the true timestamp given _height.
```

Comment:
```suggestion
                        // Cosmwasm does not expose a way to query the timestamp
                        // of a block given block height. As such, we assign migrated
                        // proposals a created timestamp of 0 so that the frontend may
                        // query for the true timestamp given `start_height`.
```